Api/ResourceInterface.php

```diff
@@ -90,4 +90,18 @@ public function getValidationGroups();
      * @return string
      */
     public function getShortName();
+
+    /**
+     * Gets the item route name of the resource (for @id generation).
+     *
+     * @return string|null
+     */
+    public function getItemRouteName();
```

Comment:
Adding new methods to an interface is a BC break. You should create a new interface containing those methods and use `instanceof` to check that this interface is implemented in the `IriConverter` class before using them.